### ISSUE TITLE
Compiler fails to catch `name` reference error

### ISSUE THREAD

**Issue by 0x80** [Microsoft/TypeScript#21718]
**TypeScript Version:**  2.7.1 and 2.6.1

**Search Terms:** 
Undefined local variable "name" not detected
ReferenceError: name is not defined

**Code**

```ts
function whatsWithName(unusedVariable: string) {
  return name;
}

whatsWithName("wut?");
```

**Expected behavior:**
Compiler telling me name is not defined.

And in addition: Compiler telling me unusedVariable is not being used.

**Actual behavior:**
Compiler says nothing. Runtime error occurs:

>ReferenceError: name is not defined

In addition I would expect the compiler to tell me that `unusedVariable` is not being used, because I have the compiler set to strict and `"noUnusedLocals": true`. Why is it not complaining about `unusedVariable`?

I've tried this with 2.6.1 and 2.7.1. Both seem to have the same behaviour.

**Playground Link:**  
https://www.typescriptlang.org/play/#src=function%20whatsWithName(unusedVariable%3A%20string)%20%7B%0D%0A%20%20%20%20console.log(name)%0D%0A%20%20return%20name%3B%0D%0A%7D%0D%0A%0D%0AwhatsWithName(%22wut%3F%22)%3B

**Related Issues:**

**Comment by mhegazy** [comment]
that is `Window.name`. see https://github.com/Microsoft/TypeScript/issues/18433 for more details.

**Comment by 0x80** [comment]
Aha! I had the dom lib included for some reason. I don't even need it. Removing it from my tsconfig causes the compiler to throw an error 👍 

I don't know why this didn't come up when I was running `tsc` on the command line. Is it included by default if you have no config?

**Comment by mhegazy** [comment]
yes. lib.d.ts (which is included by default in the absence of a `--lib` option) includes the DOM.

**Comment by 0x80** [comment]
👌 Thanks for clearing this up! 

I guess this issue can be closed then?
